src/mympi.cpp

```diff
@@ -67,9 +67,59 @@ using namespace std;
 
 namespace meep {
 
+namespace {
+
+#ifdef HAVE_MPI
+MPI_Comm mycomm = MPI_COMM_WORLD;
+#endif
+
+// comms_manager implementation that uses MPI.
+class mpi_comms_manager : public comms_manager {
```

Comment:
I guess the thinking is that you may implement other comms managers later for non-MPI backends?

Reply:
And it keeps all MPI code and awarenewss of MPI out of meep.hpp